### Import Libraries

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

from scipy import stats

### Load Cleaned Dataset

In [2]:

df = pd.read_csv("../data/cleaned_data.csv")

print(df.head())
print("\nDataset Shape:", df.shape)

  CustomerID  Tenure  MonthlyCharges  TotalCharges        Contract  \
0     C00001       6              64          1540        One year   
1     C00002      21             113          1753  Month-to-month   
2     C00003      27              31          1455        Two year   
3     C00004      53              29          7150  Month-to-month   
4     C00005      16             185          1023        One year   

      PaymentMethod PaperlessBilling  SeniorCitizen  Churn  
0       Credit Card               No              1      0  
1  Electronic Check              Yes              1      0  
2       Credit Card               No              1      0  
3  Electronic Check               No              1      0  
4  Electronic Check               No              1      0  

Dataset Shape: (500, 9)


### Descriptive Statistics by Churn Status

In [3]:


churn_statistics = (
    df.groupby("Churn")[
        ["Tenure", "MonthlyCharges", "TotalCharges"]
    ]
    .agg(["mean", "median", "std", "min", "max"])
)

print("========== DESCRIPTIVE STATISTICS BY CHURN ==========")
print(churn_statistics)

========== DESCRIPTIVE STATISTICS BY CHURN ==========
          Tenure                           MonthlyCharges                    \
            mean median        std min max           mean median        std   
Churn                                                                         
0      40.152125   42.0  18.783141   1  71     111.722595  113.0  52.130132   
1       6.000000    6.0   3.162278   1  11     129.773585  137.0  46.298630   

               TotalCharges                                  
      min  max         mean  median          std  min   max  
Churn                                                        
0      20  199  4234.577181  4096.0  2261.912833  159  7992  
1      34  198  4265.754717  4534.0  2271.043533  458  7858  


### Business Insight: Customer Characteristics by Churn

The descriptive analysis shows a major difference in customer tenure between churned and non-churned customers.

Non-churned customers have an average tenure of approximately 40.15, while churned customers have an average tenure of only 6.00. This suggests that customers are more likely to churn during the early stages of their relationship with the company.

Churned customers have higher average monthly charges (129.77) compared to non-churned customers (111.72). This suggests that monthly charges may be associated with customer churn and should be investigated further using hypothesis testing.

The average total charges are relatively similar between churned and non-churned customers. Further statistical testing will determine whether the observed differences are statistically significant.


### Hypothesis Test 1: Tenure vs Churn


In [4]:

churned_tenure = df[df["Churn"] == 1]["Tenure"]
non_churned_tenure = df[df["Churn"] == 0]["Tenure"]

t_statistic, p_value = stats.ttest_ind(
    churned_tenure,
    non_churned_tenure,
    equal_var=False
)

print("========== HYPOTHESIS TEST 1 ==========")
print("Test: Tenure Difference by Churn Status")
print("T-statistic:", t_statistic)
print("P-value:", p_value)

if p_value < 0.05:
    print("Result: Statistically Significant")
    print("Reject the Null Hypothesis.")
else:
    print("Result: Not Statistically Significant")
    print("Fail to Reject the Null Hypothesis.")

========== HYPOTHESIS TEST 1 ==========
Test: Tenure Difference by Churn Status
T-statistic: -34.53490213270181
P-value: 8.982326405691704e-130
Result: Statistically Significant
Reject the Null Hypothesis.


## Hypothesis Test 1: Tenure and Customer Churn

### Hypotheses

- Null Hypothesis (H₀): The average tenure is the same for churned and non-churned customers.
- Alternative Hypothesis (H₁): The average tenure is different for churned and non-churned customers.

### Result

The p-value is significantly smaller than the significance level of 0.05.

Therefore, the null hypothesis is rejected.

There is a statistically significant difference in customer tenure between churned and non-churned customers.

This finding supports the earlier exploratory analysis, which showed that churned customers have substantially lower average tenure. Customer tenure is therefore an important factor associated with churn in this dataset.

### Hypothesis Test 2: Monthly Charges vs Churn

In [5]:


churned_charges = df[df["Churn"] == 1]["MonthlyCharges"]
non_churned_charges = df[df["Churn"] == 0]["MonthlyCharges"]

t_statistic, p_value = stats.ttest_ind(
    churned_charges,
    non_churned_charges,
    equal_var=False
)

print("========== HYPOTHESIS TEST 2 ==========")
print("Test: Monthly Charges Difference by Churn Status")
print("T-statistic:", t_statistic)
print("P-value:", p_value)

if p_value < 0.05:
    print("Result: Statistically Significant")
    print("Reject the Null Hypothesis.")
else:
    print("Result: Not Statistically Significant")
    print("Fail to Reject the Null Hypothesis.")

========== HYPOTHESIS TEST 2 ==========
Test: Monthly Charges Difference by Churn Status
T-statistic: 2.646439327283195
P-value: 0.010079197699533079
Result: Statistically Significant
Reject the Null Hypothesis.


## Hypothesis Test 2: Monthly Charges and Customer Churn

### Hypotheses

- Null Hypothesis (H₀): The average Monthly Charges are the same for churned and non-churned customers.
- Alternative Hypothesis (H₁): The average Monthly Charges are different for churned and non-churned customers.

### Result

The p-value of approximately 0.0101 is smaller than the significance level of 0.05.

Therefore, the null hypothesis is rejected.

There is a statistically significant difference in Monthly Charges between churned and non-churned customers.

The descriptive analysis showed that churned customers have higher average Monthly Charges (129.77) compared to non-churned customers (111.72).

This suggests that Monthly Charges are associated with customer churn in this dataset. However, this statistical relationship does not prove that higher Monthly Charges directly cause customer churn.

### Hypothesis Test 3: Total Charges vs Churn

In [6]:


churned_total_charges = df[df["Churn"] == 1]["TotalCharges"]
non_churned_total_charges = df[df["Churn"] == 0]["TotalCharges"]

t_statistic, p_value = stats.ttest_ind(
    churned_total_charges,
    non_churned_total_charges,
    equal_var=False
)

print("========== HYPOTHESIS TEST 3 ==========")
print("Test: Total Charges Difference by Churn Status")
print("T-statistic:", t_statistic)
print("P-value:", p_value)

if p_value < 0.05:
    print("Result: Statistically Significant")
    print("Reject the Null Hypothesis.")
else:
    print("Result: Not Statistically Significant")
    print("Fail to Reject the Null Hypothesis.")

========== HYPOTHESIS TEST 3 ==========
Test: Total Charges Difference by Churn Status
T-statistic: 0.09453830507131283
P-value: 0.9249730721139848
Result: Not Statistically Significant
Fail to Reject the Null Hypothesis.


## Hypothesis Test 3: Total Charges and Customer Churn

### Hypotheses

- Null Hypothesis (H₀): The average Total Charges are the same for churned and non-churned customers.
- Alternative Hypothesis (H₁): The average Total Charges are different for churned and non-churned customers.

### Result

The p-value of approximately 0.925 is much greater than the significance level of 0.05.

Therefore, we fail to reject the null hypothesis.

There is no statistically significant difference in Total Charges between churned and non-churned customers in this dataset.

This suggests that Total Charges alone may not be an important factor associated with customer churn in this analysis.

## 95% Confidence Interval for Monthly Charges

In [7]:


monthly_charges = df["MonthlyCharges"]

mean = monthly_charges.mean()
std_error = stats.sem(monthly_charges)

confidence_interval = stats.t.interval(
    0.95,
    len(monthly_charges) - 1,
    loc=mean,
    scale=std_error
)

margin_of_error = (
    confidence_interval[1] - confidence_interval[0]
) / 2

print("========== 95% CONFIDENCE INTERVAL ==========")
print("Mean Monthly Charges:", mean)
print("Margin of Error:", margin_of_error)
print("Lower Bound:", confidence_interval[0])
print("Upper Bound:", confidence_interval[1])
print(
    f"95% Confidence Interval: "
    f"({confidence_interval[0]:.2f}, {confidence_interval[1]:.2f})"
)

========== 95% CONFIDENCE INTERVAL ==========
Mean Monthly Charges: 113.636
Margin of Error: 4.5514176289484425
Lower Bound: 109.08458237105155
Upper Bound: 118.18741762894844
95% Confidence Interval: (109.08, 118.19)


## 95% Confidence Interval for Monthly Charges

The average Monthly Charges in the dataset are approximately 113.64.

The calculated margin of error is approximately ±4.55.

The 95% confidence interval for the average Monthly Charges is:

**(109.08, 118.19)**

Based on this sample, the underlying population mean Monthly Charges is estimated to lie within this interval with 95% confidence.

# Final Business Insights

## Key Findings

### 1. Overall Customer Churn

The overall customer churn rate is 10.6%, with 53 out of 500 customers leaving the company.

### 2. Customer Tenure

Customer tenure is strongly associated with churn.

Non-churned customers have an average tenure of approximately 40.15, while churned customers have an average tenure of only 6.00.

The hypothesis test confirmed that this difference is statistically significant.

This indicates that new and early-stage customers represent an important high-risk group for churn.

### 3. Contract Type

Month-to-month customers have the highest churn rate at approximately 20.59%.

Customers with longer-term contracts have considerably lower churn rates, particularly one-year contract customers at approximately 4.30%.

This suggests that encouraging customers to move toward longer-term contracts may help improve retention.

### 4. Monthly Charges

Churned customers have higher average Monthly Charges (129.77) compared to non-churned customers (111.72).

The hypothesis test showed that this difference is statistically significant.

Monthly Charges may therefore be an important factor associated with customer churn.

### 5. Payment Method

Credit Card customers have the highest observed churn rate at approximately 13.48%, followed by Electronic Check customers at approximately 11.04%.

Bank Transfer customers have the lowest churn rate at approximately 6.92%.

Payment method may be associated with differences in customer churn behavior.

### 6. Total Charges

The analysis found no statistically significant difference in Total Charges between churned and non-churned customers.

Therefore, Total Charges alone may not be a strong factor associated with churn in this dataset.

# Business Recommendations

## Recommendation 1: Focus on Early Customer Retention

Customers with low tenure appear to have a higher risk of churn.

The company should create onboarding programs, welcome offers, and early customer engagement campaigns to improve retention during the first few months.

## Recommendation 2: Encourage Longer-Term Contracts

Month-to-month customers have the highest churn rate.

The company can offer discounts, loyalty benefits, or special incentives to encourage customers to move from month-to-month contracts to one-year or two-year contracts.

## Recommendation 3: Review Pricing for High-Risk Customers

Churned customers have higher average Monthly Charges.

The company should evaluate whether high-cost customers would benefit from personalized pricing plans, discounts, or improved service value.

## Recommendation 4: Investigate Payment Experience

Customers using different payment methods show different churn rates.

The company should investigate customer experience and potential issues related to payment methods and improve convenience where necessary.

## Recommendation 5: Develop a Customer Churn Monitoring System

The company should regularly monitor important churn indicators such as:

- Customer tenure
- Monthly charges
- Contract type
- Payment method
- Customer engagement

Customers showing multiple risk indicators could be targeted with proactive retention campaigns.

# Business Implementation Plan

## Phase 1: Identify High-Risk Customers

Identify customers with:

- Low tenure
- Month-to-month contracts
- Higher monthly charges
- Other observed risk indicators

## Phase 2: Targeted Retention Campaigns

Develop personalized retention campaigns such as:

- Welcome offers for new customers
- Discounts for high-value customers
- Contract upgrade incentives
- Loyalty rewards

## Phase 3: Monitor Results

Track the following metrics:

- Monthly churn rate
- Customer retention rate
- Contract conversion rate
- Customer satisfaction
- Retention campaign performance

## Phase 4: Continuous Improvement

Regularly analyze customer data and evaluate whether retention strategies are reducing churn.

The business should update its retention strategy based on new data and customer behavior.

# Conclusion

This capstone project conducted an end-to-end business analysis of customer churn using a dataset containing 500 customer records.

The project included data validation, data cleaning, exploratory data analysis, statistical analysis, hypothesis testing, confidence intervals, and business recommendations.

The analysis identified customer tenure, contract type, and monthly charges as important factors associated with customer churn.

The strongest finding was that churned customers had substantially lower average tenure compared to non-churned customers.

Based on these findings, the business should prioritize early customer engagement, encourage longer-term contracts, review pricing strategies, and implement proactive churn monitoring.

The results of this project demonstrate how data analysis can transform raw business data into actionable insights and practical business recommendations.